README.md

```diff
@@ -39,3 +63,23 @@ optional arguments:
                         to the workspace's devel space! Default is to use the current catkin
                         workspace layers ("sourced").
 ```
+
+### Example
+
+In order to create eclipse project files for a ros package called **my_pkg** that lives in a 
+catkin workspace then do the following:
+- CD into the top level directory of your workspace
+
+- Run eclipsify
+    ```
+    eclipsify my_pkg -O project/my_pkg
```

Comment:
This would be an advanced usage. Normally you just do 
* `eclipsify my_pkg` : Generated project files go into  `<workspace>/devel/share/my_pkg/eclipse/` and the my_pkg source folder gets _emph_ linked into the project (eclipse's linked folder) at least with the default project files templates. The same happens if one specifies `-O` without the `-s`.
* `eclipsify -s my_pkg` : Generated project files go into the source folder of my_pkg (e.g. `src/my_pkg`). And no linked folder gets created.

The latter (I would normally recommend) is required if the user wants to use EGit  (git through eclipse; https://www.eclipse.org/egit/). There is also a difference in terms of link

Reply:
- I do prefer to keep the eclipse object files separate from the directories managed by catkin as cleaning the workspace won't require you to recreate the project files.  
- I can certainly add these advanced cases even though I'm not sure I understand how eclipse deals with linked folders vs unlinked ones (-s case).  
- Thanks for the feedback, this is a great tool, really like the fact that it does use the catkin build system.

Reply:
I'm happy that you like this little tool :). Thanks for the feedback.

I'm a bit confused now. Where is `project/my_pkg` in your example? Isn't it the source folder of `my_pkg` within the source folder of your catkin workspace (e.g. `catkin_ws/src`)? 

**If so,** then providing `-s` only is doing almost exactly the same as what you suggest, while being a bit shorter to write. The only difference would then be that in addition it adds the `-D IN_SOURCE=1` to the cpp preprocessor internally used to instantiate the project file templates. And that has currently the effect defined here: https://github.com/ethz-asl/eclipsify/blob/master/src/eclipsify_lib/templates/linux2/.project#L27-L35 
Informally it ads a link to the projects source folder into the eclipse-project unless `IN_SOURCE` is defined, which represents the fact that the eclipse-project file itself is already within the source folder of the project (so no link needed!).

**If not so**, the we should add all variants as examples and explain the difference.

Reply:
@HannesSommer I create those project files in the catkin workspace top level directory, so my catkin ws looks like this more or less
```
catkin_ws
  /src
  /build
  /devel
  /projects //-->  this is where I put my eclipse project files
     /mypkg_1/.cproject, .project, .settings
     /mypkg_2/ ... 
     .
     .
```

Reply:
I prefer keeping any IDE specific project files outside the source folders as that avoids the risk of accidentally adding them to the repository.  In fact, I'd prefer that eclipsify would not save then in the **src** directory by default when no special options are passed to it.